### PR TITLE
Documentaton of Integration, NoxSimpleType and UserInterface classes.

### LINKED ISSUE #44: [Nox.Solution] Documentation of classes

### PR DESCRIPTION
**Pull request by Mornevanzyl**
- Documented new Integration classes, i.e. database, file, http and message queue
- Documented NoxSimpleType
- Documented UserInterfaceType

### CODE REVIEW

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs [hunk 1]
```diff
@@ -2,22 +2,59 @@
 
 namespace Nox.Solution
 {
-
+    [Title("UI display options for this attribute.")]
+    [Description("Specify how this attribute is rendered on the Nox user interface. Configuration options include label, icon, input/output masks, help hints among other.")]
```
Comment (andresharpe) [inline src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:6]: On any interface... you can remove "Nox" - someone may build better interfaces using these definitions

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs [hunk 2]
```diff
@@ -2,22 +2,59 @@
...
     [AdditionalProperties(false)]
     public class TypeUserInterface : DefinitionBase
     {
+        [Title("A descriptive label for this attribute on the UI.")]
+        [Description("The text label rendered on the user interface alongside the NoxTextInput component.")]
         public string? Label { get; internal set; }
+
+        // Pending implementation details
```
Comment (andresharpe) [inline src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:14]: let's do these. this project doesn't depend on implementation

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs [hunk 3]
```diff
@@ -2,22 +2,59 @@
...
         public string? TranslationId { get; internal set; }
+
+        // Pending implementation details
```
Comment (andresharpe) [inline src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:17]: ditto

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs [hunk 4]
```diff
@@ -2,22 +2,59 @@
...
+        [Title("The position of the attribute icon on the UI.")]
+        [Description("The postition of the attribute icon relative to the text input area on the user interface, e.g. begin or end.")]
         public IconPosition? IconPosition { get; internal set; } = global::Nox.IconPosition.Begin;
+
+        [Title("A mask expression used to validate and control text input.")]
+        [Description("A string of characters that indicates the format of valid input values. Examples include '*?@?*.?*' for an email address and '(000) 000-0000' for a seven digit phone number with mandatory area code.")]
         public string? InputMask { get; internal set; }
+
+        [Title("A mask expression used to validate and control text output.")]
+        [Description("A string of characters that indicates the format of valid output. Used to handle validation triggers for onSubmit and onChange events of the control.")]
```
Comment (andresharpe) [inline src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:33]: May not be text...use "value's output"

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs [hunk 5]
```diff
@@ -2,22 +2,59 @@
...
         public string? OutputMask { get; internal set; }
+
+        [Title("The regular expression against which input is validated.")]
+        [Description("A regex string that indicates the format of valid input. Examples include '/^[^@\\s]+@[^@\\s]+\\.[^@\\s]+$/gm' for email address and '/^(\\([0-9]{3}\\) |[0-9]{3}-)[0-9]{3}-[0-9]{4}$/gm' for a seven digit phone number with mandatory area code.")]
```
Comment (andresharpe) [inline src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:37]: We won't use a mask for email output. numbers, currencies, and password will be masked.

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs [hunk 6]
```diff
@@ -2,22 +2,59 @@
...
         public string? Regex { get; internal set; }
+
+        [Title("The UI PageGroup ID.")]
+        [Description("Used to associate component with a PageGroup ID on the user interface.")]
```
Comment (andresharpe) [inline src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:41]: ID -> identifier

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs [hunk 7]
```diff
@@ -2,22 +2,59 @@
...
         public string? PageGroup { get; internal set; }
+
+        [Title("The UI FieldGroup ID.")]
```
Comment (andresharpe) [inline src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:44]: UI -> user interface. Avoid acronyms anywhere

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs [hunk 8]
```diff
@@ -2,22 +2,59 @@
...
+        [Description("Used to associate component with a FieldGroup ID on the user interface.")]
         public string? FieldGroup { get; internal set; }
+
+        [Title("FieldGroup tab order for this UI component.")]
```
Comment (andresharpe) [inline src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:48]: The order in which this attribute will be displayed or edited.

**Review by andresharpe — CHANGES_REQUESTED** [review]

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs (hunk 1 above)
Comment (Mornevanzyl) [reply src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:6]: Done, have removed UI acronyms in favour of full text.

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs (hunk 4 above)
Comment (Mornevanzyl) [reply src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:33]: Done.

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs (hunk 6 above)
Comment (Mornevanzyl) [reply src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:41]: Done

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs (hunk 7 above)
Comment (Mornevanzyl) [reply src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:44]: Done.

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs (hunk 2 above)
Comment (Mornevanzyl) [reply src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:14]: Done.

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs (hunk 3 above)
Comment (Mornevanzyl) [reply src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:17]: Done.

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs (hunk 8 above)
Comment (Mornevanzyl) [reply src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:48]: [Title("Display/edit focus order within a FieldGroup on a user interface.")]
        [Description("The order within a group of user interface components (FieldGroup) in which this attribute will receive focus for display or edit purpose.")]

File: src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs (hunk 5 above)
Comment (Mornevanzyl) [reply src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs:37]: Removed email examples for InputMask and Regex.

**Comment by andresharpe** [comment]
Merged with #56

**Pull request closed without merging**